recipes-bsp/firmware-pack/bootparameter/bootparameter.c

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+/*
+ * Copyright (C) 2021 Renesas Electronics Corp.
+ * Author: Biju Das <biju.das.jz@bp.renesas.com>
+*/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+
+off_t fsize(const char *filename)
+{
+	struct stat st;
+	if (stat(filename, &st) == 0)
+		return st.st_size;
+	return -1;
+}
+
+int main(int argc, char *argv[])
+{
+	int size, i = 0;
+	char val = 0xff;
+	if (argc <= 2) { /* argc should be 2 for correct execution */
+		/* We print argv[0] assuming it is the program name */
+		printf("usage: %s bl2_filename output_file_name\n", argv[0]);
+		return -1;
+	} else {
+		// We assume argv[1] is a filename to open
+		FILE *fptr;
+		fptr = fopen(argv[2], "wb");
+		/* fopen returns 0, the NULL pointer, on failure */
+		if (fptr == 0)	{
+			printf("Could not open file\n");
+		}
+		int fd = fileno(fptr);
+		size = fsize(argv[1]);
+		printf("size is %x\n",size);
+		size = (size + 3) & (~0x3);
+		printf("Aligned size is %x\n",size);
+
+		fwrite(&size, 4, 1, fptr);
+		for(i=0;i<506; i++)
```

Comment:
Using 506 because the boot parameter program writes data to sector 1, converting it into a Master Boot Record (MBR) and sector size is 512bytes.

- Reference link: https://renesasgroup.sharepoint.com/sites/team_SOM/_layouts/15/Doc.aspx?sourcedoc={a5dfca3e-3da8-4eef-91b9-f559be4aa8ae}&action=edit&wd=target%28Knowledge%20Sharing.one%7C4952f301-92e5-41dd-b8f0-3ce16480fe1a%2FRZ%20common%20-%20Boot%20parameters%7C9ee6a7e2-6138-4ad2-a3a2-4df4b315a3f4%2F%29&wdorigin=NavigationUrl

// Write 4 byte at the beginning of the sector 1. 
fwrite(&size, 4, 1, fptr); 
// Write a 2-bytes signature word or end of sector marker, which is always set to 0x55AA 
 val = 0x55; 
 fwrite(&val, 1, 1, fptr); 
 val = 0xAA; 
 fwrite(&val, 1, 1, fptr); 

-  So, the remaining total bytes are 512 - 2 -4 =506 bytes.If file sizes are less than 506 or greater than 506, booting is failed because sector size is not correct.
- Updated in the commit: 159cdfe195d266aa671000a357e36c4b7fa95f1a